src/main/java/org/broadinstitute/hellbender/tools/coveragemodel/LinearSpaceBlock.java

```diff
@@ -0,0 +1,89 @@
+package org.broadinstitute.hellbender.tools.coveragemodel;
+
+import org.broadinstitute.hdf5.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+
+import javax.annotation.Nonnull;
+import java.io.Serializable;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+/**
+ * This class represents a key for a contiguous block of elements in an indexed linear space.
+ *
+ * The begin index is include
+ * The end index is exclusive
+ *
+ * @author Mehrtash Babadi &lt;mehrtash@broadinstitute.org&gt;
+ */
+public final class LinearSpaceBlock implements Serializable {
```

Comment:
I renamed it to `LinearlySpacedIndexBlock`. It is descriptive and hopefully not too verbose. I also made it package-local.